mymake.cpp

```diff
@@ -83,16 +83,18 @@ int main(int argc, char **argv) {
   set_linux();
 #endif
   for(string fname: {"Makefile.loc", "Makefile.simple", "Makefile"})
-    if(file_exists(fname))
+    if(file_exists(fname)) {
       system("make -f " + fname + " language-data.cpp autohdr.h savepng.o");
+      break;
+      }
   for(int i=1; i<argc; i++) {
     string s = argv[i];
     if(s.substr(0, 2) == "-D") {
       opts += " " + s;
       obj_dir += "/";
       setdir += "../";
       for(char c: s) 
-        if(c == '=' || c == '-' || c == '/') obj_dir += "_"; 
+        if(c == '=' || c == '-' || c == '/' || c == '"' || c == '\\') obj_dir += "_"; 
```

Comment:
The only really dangerous character on POSIX is `/`; but the shell might care about any of `'"$!()[]{};%*&<>`. To be safe I'd just replace these two lines with `obj_dir += (isalnum(c) ? c : '_');`

Reply:
Makes sense. Probably better to replace more "problematic" characters, since there is also Windows, for example, under which I'm currently trying to coerce `mymake` into working.
There is also an `if` to pretty much the same effect here:
https://github.com/zenorogue/hyperrogue/blob/0813fb65ca565cadf7baa971615a643e06be1357/mymake.cpp#L120
I guess the same reasoning would apply there? And at any rate, now that @zenorogue has merged the PR, what would be the appropriate thing to do to make that change?

Reply:
I have pushed the change where I replaced the check with `!isalnum(c)` like @Quuxplusone suggested. I think making mymake work on Windows should be another PR. Thanks!

Reply:
Yeah, I'll create a PR if I succeed (has not yet happened). OK, will now delete the branch then.